include/silo/storage/reference_genomes.h

```diff
@@ -11,27 +11,31 @@
 namespace silo {
 
 struct ReferenceGenomes {
-   std::map<std::string, std::vector<Nucleotide::Symbol>> nucleotide_sequences;
-   std::map<std::string, std::vector<AminoAcid::Symbol>> aa_sequences;
-   std::map<std::string, std::string> raw_nucleotide_sequences;
-   std::map<std::string, std::string> raw_aa_sequences;
+   std::vector<std::string> nucleotide_sequence_names;
+   std::vector<std::string> aa_sequence_names;
+   std::vector<std::vector<Nucleotide::Symbol>> nucleotide_sequences;
+   std::vector<std::vector<AminoAcid::Symbol>> aa_sequences;
```

Comment:
Is it really a good idea to store sequence names and sequences separately? It bloats the code in several places and detaches information that actually belongs together (and thus should be attached).

Reply:
I introduced this change rather late while working on the PR. This cleaned up the code a lot in many places